ipfsapi/encoding.py

```diff
@@ -18,21 +22,62 @@
 class Encoding(object):
     """Abstract base for a data parser/encoder interface.
     """
+    __metaclass__ = abc.ABCMeta
 
-    def parse(self, string):
-        """Parses string into corresponding encoding.
+    @abc.abstractmethod
+    def parse_partial(self, raw):
```

Comment:
Yes, you call `parse_partial` as often as you want (collecting the results), then call `parse_finalize` to process and obtain any stuff that may be remaining the `encoding`'s buffers. For JSON that's just error handling (there is no reliable way to determine whether an error happened because of EOF or not).